packages/hdwallet-provider/src/index.ts

```diff
@@ -37,13 +58,15 @@ class HDWalletProvider {
   public engine: ProviderEngine;
 
   constructor(
-    mnemonic: string | string[],
+    mnemonic: string | string[] | ConstructorOptions,
```

Comment:
can we call this `mnemonicOrOptions` and only assign to a variable `mnemonic` when we're sure it's a mnemonic?